src/client/Microsoft.Identity.Client/ApiConfig/AbstractConfidentialClientAcquireTokenParameterBuilder.cs

```diff
@@ -12,19 +13,21 @@
 
 namespace Microsoft.Identity.Client
 {
-#if !ANDROID_BUILDTIME && !iOS_BUILDTIME && !WINDOWS_APP_BUILDTIME && !MAC_BUILDTIME // Hide confidential client on mobile platforms
-
     /// <summary>
     /// Base class for confidential client application token request builders
     /// </summary>
     /// <typeparam name="T"></typeparam>
+#if !SUPPORTS_CONFIDENTIAL_CLIENT
```

Comment:
I created a few new flags: 

- SUPPORTS_CONFIDENTIAL_CLIENT aka all platforms except mobile
- SUPPORTS_CUSTOM_CACHE - aka all platforms except mobile
- SUPPORTS_BROKER aka Android, iOS and UWP